### PR TITLE
feat: status message

### PR DESCRIPTION
**Pull request by simonheys**
Adds the ability to display a message in the extension

* Fullscreen 'danger' message, can be closed but not dismissed completely

https://user-images.githubusercontent.com/175607/181522062-00c1de0e-8e38-49e6-824e-73f48f37f9a2.mov

* Warn banner

https://user-images.githubusercontent.com/175607/181522078-36867189-a914-4d56-a286-9bb2d5a46d0d.mov

* Info banner

https://user-images.githubusercontent.com/175607/181522098-c33f986c-6d5b-4ac3-b782-39606aa5eafc.mov

* Storybook
Screenshot 2022-07-28 at 14 46 46
Screenshot 2022-07-28 at 14 46 51
Screenshot 2022-07-28 at 14 47 01
Screenshot 2022-07-28 at 14 47 07
Screenshot 2022-07-28 at 14 47 12

### CODE REVIEW

File: packages/extension/package.json [hunk 1]
```diff
@@ -89,6 +90,7 @@
     "react-hook-form": "^7.33.0",
     "react-measure": "^2.5.2",
     "react-router-dom": "^6.0.1",
+    "semver": "^7.3.7",
```
Comment (simonheys) [inline packages/extension/package.json:93]: Used to compare versions - we only import two specific functions

File: packages/extension/src/shared/statusMessage/storage.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import { KeyValueStorage } from "../storage"
+import { IStatusMessageStorage } from "./types"
+
+export const statusMessageStore = new KeyValueStorage<IStatusMessageStorage>(
+  {},
+  {
+    namespace: "status:statusMessage",
```
Comment (simonheys) [inline packages/extension/src/shared/statusMessage/storage.ts:7]: Doesn't feel like a 'core:...' feature - but ... what should it be?

File: packages/extension/src/ui/components/Button.tsx [hunk 3]
```diff
@@ -23,38 +32,71 @@ export const getVariantColor =
...
   ({ variant }: IButton) => {
     switch (variant) {
-      case "warn":
+      case "danger":
+        return hover
+          ? colord(theme.red1).lighten(0.075).toRgbString()
+          : disabled
+          ? colord(theme.red1).alpha(0.5).toRgbString()
+          : theme.red1
+      case "warn-high":
```
Comment (simonheys) [inline packages/extension/src/ui/components/Button.tsx:41]: There are two 'warn' colours in the designs. Hopefully will be rationalised at some future point.

File: packages/extension/src/ui/features/statusMessage/useStatusMessage.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    ARGENT_X_STATUS_URL,
+    fetcher,
+    {
+      refreshInterval: 5 * 60 * 60 * 1000 /** 5 minutes */,
+    },
+  )
+  return statusMessage
+}
+
+// Uncomment the functions below to test messages locally
```
Comment (simonheys) [inline packages/extension/src/ui/features/statusMessage/useStatusMessage.ts:40]: I felt this was useful to leave in - allows you to try different messages locally using test fixures

File: packages/extension/src/shared/api/constants.ts [hunk 5]
```diff
@@ -28,3 +28,5 @@ export const ARGENT_TRANSACTION_REVIEW_STARKNET_URL =
         "transactions/review/starknet",
       )
     : undefined
+
+export const ARGENT_X_STATUS_URL = process.env.ARGENT_X_STATUS_URL as string
```
Comment (janek26) [inline packages/extension/src/shared/api/constants.ts:32]: can be undefined, no?

File: packages/extension/src/shared/statusMessage/storage.ts (hunk 2 above)
Comment (janek26) [reply packages/extension/src/shared/statusMessage/storage.ts:7]: maybe we should introduce `service:` namespace for all things that are fully optional ?

**Review by janek26 — APPROVED** [review]
I love all the stories and tests! I should start using it more once I get back to feature development

**Pull request merged**